mws/apis/subscriptions.py

```diff
@@ -17,76 +18,145 @@ class Subscriptions(MWS):
     Docs:
     http://docs.developer.amazonservices.com/en_US/subscriptions/index.html
     """
-    URI = '/Subscriptions/2013-07-01'
-    VERSION = '2013-07-01'
+    URI = "/Subscriptions/2013-07-01"
+    VERSION = "2013-07-01"
     NAMESPACE = "{http://mws.amazonaws.com/Subscriptions/2013-07-01}"
 
     # TODO include a helper method for configuring and saving a destination to the object with a keyname
     # This might cut down on some time setting up all the values for the destination for each call,
     # particularly if someone needs to make several calls at once for the same destination.
 
-    def register_destination(self):
+    def register_destination(self, marketplace_id, delivery_channel="SQS", attribute_list=None):
         """
         Specifies a new destination where you want to receive notifications.
 
         Docs:
         http://docs.developer.amazonservices.com/en_US/subscriptions/Subscriptions_RegisterDestination.html
+
+        delivery_channel: Currently only supports SQS
+        attribute_list: example [{"sqsQueueUrl" : "https://sqs.us-east-1.amazonaws.com/51471EXAMPLE/mws_notifications"}]
```

Comment:
Example will produce:

    Destination.AttributeList.member.1.sqsQueueUrl=https://sqs.us-east-1.amazonaws.com/51471EXAMPLE/mws_notifications

MWS expects, per example request [here](http://docs.developer.amazonservices.com/en_US/subscriptions/Subscriptions_RegisterDestination.html):

    Destination.AttributeList.member.1.Key=sqsQueueUrl
    Destination.AttributeList.member.1.Value=https://sqs.us-east-1.amazonaws.com/51471EXAMPLE/mws_notifications

---

I suggest a helper function (outside the class) that can make a simple conversion from key-value pairs in a single dict to the proper `AttributeList` output:

```python
def parse_attributes(attributes):
    if not attributes or not isinstance(attributes, dict):
        # Return empty dict so it can easily pass to `data.update()`
        return {}
    attribute_list = []
    for key, val in attributes:
        attribute_list.append({
            'Key': key,
            'Value': val,
        })
    params = utils.enumerate_keyed_param("Destination.AttributeList.member", attribute_list)
    return params
```

Then the request methods can accept a simple dict object as `attributes` and call the following:

```python
def register_destination(self, marketplace_id, delivery_channel="SQS", attributes=None):
    ...
    data.update(parse_attributes(attributes))
```

This makes use of the method simpler for the end user, I think. Now passing `attributes = {"sqsQueueUrl" : "https://sqs.us-east-1.amazonaws.com/51471EXAMPLE/mws_notifications"}` should work properly.

Reply:
@Bobspadger I see your last comment saying thing are working in production. Can you confirm which inputs you're using that are working? I only have the MWS docs to go on, so if the example this is noting actually works, then that's worth noting.

Reply:
@GriceTurrble for my tests I was passing in 
```python
att_list = [{
     "Key": "sqsQueueUrl",
     "Value":"https://sqs.eu-west-2.amazonaws.com/378051892504/Amazon_MWS_Notify"
}]
```

and called is at:

```python
new_sub = subscriptions.register_destination(MARKETPLACE_ID, delivery_channel='SQS', attribute_list=att_list )```

which gives the required output.
I've done it this way as the docs show as enumerated with key value, so rather than only have those possible options I passed in the dict with the key value

Reply:
Understood.

From what I see in the docs, they will only expect "Key" and "Value" (unless they anticipate adding more options in future). We can abuse that as I suggested so our end users only have to pass a dict, rather than a list of dicts with specific keys.

Reply:
Amazon appear to suggest that there may be more queue options coming, which is why the enumerate the key/value pair.

Do we build this to accept a list of dicts, or just a single dict.

Do we second guess amazon, or just work with what there currently is (example SQS is the only accepted delivery, even though they give us the option of specifying a delivery method)